flask_appbuilder/security/sqla/apis/group/api.py

```diff
@@ -0,0 +1,213 @@
+from flask import request
+from flask_appbuilder import ModelRestApi
+from flask_appbuilder.api import expose, safe
+from flask_appbuilder.const import API_RESULT_RES_KEY
+from flask_appbuilder.models.sqla.interface import SQLAInterface
+from flask_appbuilder.security.decorators import permission_name, protect
+from flask_appbuilder.security.sqla.apis.group.schema import (
+    GroupPostSchema,
+    GroupPutSchema,
+)
+from flask_appbuilder.security.sqla.models import Group, Role, User
+from marshmallow import ValidationError
+from sqlalchemy.exc import IntegrityError
+
+
+class GroupApi(ModelRestApi):
+    resource_name = "security/groups"
+    openapi_spec_tag = "Security Groups"
+    class_permission_name = "Group"
+    datamodel = SQLAInterface(Group)
+    allow_browser_login = True
+
+    list_columns = ["id", "name", "label", "description", "roles", "users"]
+    show_columns = list_columns
+    edit_columns = ["name", "label", "description", "users", "roles"]
+    add_columns = edit_columns
+    search_columns = list_columns
+
+    add_model_schema = GroupPostSchema()
+    edit_model_schema = GroupPutSchema()
+
+    @expose("/", methods=["POST"])
+    @protect()
+    @safe
+    @permission_name("post")
+    def post(self):
+        """Create new group
+        ---
+        post:
+          requestBody:
+            description: Model schema
+            required: true
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/{{self.__class__.__name__}}.post'
+          responses:
+            201:
+              description: Group created
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      result:
+                        $ref: '#/components/schemas/{{self.__class__.__name__}}.post'
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        try:
+            item = self.add_model_schema.load(request.json)
+            model = Group()
+            roles = []
+            users = []
+
+            for key, value in item.items():
+                if key == "roles":
+                    for role_id in value:
+                        role = (
+                            self.datamodel.session.query(Role)
+                            .filter(Role.id == role_id)
+                            .one_or_none()
+                        )
+                        if not role:
+                            return self.response_400(
+                                message={
+                                    "roles": [f"Role with ID {role_id} does not exist."]
+                                }
+                            )
```

Comment:
We could avoid making 1 query for each role or user, also would be nice to add and helper method for this to improve readability:

``` python
def _fetch_entities(self, model_class, ids):
    if not ids:
        return []
    return (
        self.datamodel.session.query(model_class)
        .filter(model_class.id.in_(ids))
        .all()
    )
```